New documentation structure

joshring opened the pull request:
- Experimental new structure and new ordering
- Worked on installation instructions
- Worked on comments
- Adding annotations to code blocks
- Adding some improvements to c interop section
- Smoothing out the initial run through of their first program and making sure people get the information about installing C3
- Fixing links to go with new structure, including changelog, specification etc
- Formatting of code
- Minor clarifications


If you want to "follow allow" there is a staging env which i've been keeping up to date with progress:
https://experimental-site-testing-only.github.io/

![image](https://github.com/user-attachments/assets/07c6840b-384d-46bd-8689-f0782a8eb50d)


![image](https://github.com/user-attachments/assets/76a608e4-029f-4420-bea0-9d565c8a3e38)

![image](https://github.com/user-attachments/assets/100eb61f-d03c-43f6-b5f6-dc03405e9088)

joshring committed:
third attempt

joshring committed:
added redirects to common pages in the old structure
corrected links on frontpage
made navbar more obvious on hover, not less
added discord to navbar rather than dropdown

joshring committed:
numbered list number restored

lerno commented on src/content/docs/Build Your Project/build-commands.md:
```diff
@@ -41,12 +55,11 @@ It is also possible to give the path to a custom template.
-$ tree .
-.
-└── hello_world
+hello_world
```
This removed the example where it was clearly the output of the `tree` command?

lerno commented on src/content/docs/Build Your Project/build-commands.md:
````diff
@@ -65,42 +78,56 @@ $ tree .
 
-`clean`
+```bash
+clean
````
This should not be marked "bash" same with some other stuff above and below.

joshring replied:
No problem.

I added c3c to the start of the commands so they were more obvious what they were for

Also found 
`c3c docs [<target>]` 
needed to be changed to:
`c3c directives [<target>]`

lerno replied:
Directives??

joshring replied:
![image](https://github.com/user-attachments/assets/b316871d-f992-427a-8891-bcdaffcd2e81)

lerno commented on src/content/docs/FAQ/changesfromc.md:
```diff
@@ -113,29 +115,31 @@ the code is harder to understand as well. The replacements together with `defer`
 ##### Locals variables are implicitly zeroed
 
-In C global variables are implicitly zeroed out, but local variables aren't. In C3 local variables are zeroed out by default, but may be explicitly undefined to get the C behaviour.
+In C global variables are implicitly zeroed out, but local variables aren't. In C3 local variables are zeroed out by default, but may be *explicitly* undefined if you wish to match the C behaviour.
```
> be *explicitly* undefined
can be changed to
> be *explicitly* undefined (using the `@noinit` attribute)

lerno commented on src/content/docs/Generic Programming/anyinterfaces.md:
```diff
@@ -118,21 +127,25 @@ implement interfaces is possible but does not provide compile time checks.
+fn String Baz.to_new_string(Baz baz, Allocator allocator) @dynamic
+{
+    return string::printf("Baz(%d)", baz.x, .allocator = allocator);
+}
```
Should be `allocator: allocator` not `.allocator = allocator)`

lerno commented on src/content/docs/Language Fundamentals/naming.md:
```diff
@@ -2,18 +2,18 @@
 ---
 
-C3 introduces fairly strict naming rules to reduce ambiguity. 
+C3 introduces fairly strict naming rules to reduce ambiguity, make the language compile faster and simplify the compiler. 
```
Change to:
C3 introduces fairly rigid naming rules to reduce ambiguity and make the language easy to parse for tools.

lerno commented on src/content/docs/Language Overview/types.md:
```diff
@@ -414,7 +415,8 @@ struct Position
 
-enum State : int (String desc, bool active, Position pos)
+enum State : int (String desc, bool active, Position pos)sidebar:
+    order: 135
```
Something went wrong here.

joshring committed:
fixes from comments

joshring committed:
Added roadmap based on recent discord discussion
Added some details to the contribute page
improved the wording in a few places

lerno commented on src/content/docs/Language Overview/types.md:
```diff
@@ -415,7 +415,7 @@ struct Position
 
-enum State : int (String desc, bool active, Position pos)sidebar:
+enum State : int (String desc, bool active, Position pos)
     order: 135
```
This "order" is wrong too.

lerno commented on src/content/docs/Build Your Project/build-commands.md:
````diff
@@ -88,46 +90,47 @@ The built-in templates define two targets: `debug` (which is the default) and `r
-```bash
-bench [target]
+```
+c3c bench [target]
````
If you write `c3c` first though, then it can be "bash"

lerno commented on src/content/docs/Build Your Project/build-commands.md:
````diff
@@ -88,46 +90,47 @@ The built-in templates define two targets: `debug` (which is the default) and `r
+c3c directives [<target>]
+```
+                              
+Generate documentation for the target.
````
No, it's docs. You generate documentation for the target. Note that this one is not complete yet and should be noted as such.

joshring committed:
capitalisation

joshring committed:
leftover characters removed

joshring committed:
addressing comments, improving dark mode closes https://github.com/c3lang/c3-web/pull/43

joshring committed:
hover colour fix

joshring committed:
updating from comments
darkmode colours
frontpage download button, making more obvious while still in-keeping with the theme

joshring committed:
improving colours

joshring committed:
colours update

joshring committed:
colour update

joshring committed:
- updating links to use consistent format which is easier to find and replace so easier to restructure
- updating frontpage with new download button custom extra downloads button and added keyboard navigation support
- new get started button matching download button
- updates matching comments to some content, more to do

joshring committed:
content update
restructure optionals section

joshring committed:
updating defer

joshring committed:
latest comments

joshring committed:
removing reminder as moved section

joshring committed:
capitalisation and naming of pages tweaks

joshring committed:
avoid scrolling

joshring committed:
An -> The

joshring committed:
neatening up download links

joshring committed:
headings and some minor organisation

joshring committed:
resized

joshring committed:
tweaked wording on examples
